src/FbgemmFP16UKernelsAvx2.cc

```diff
@@ -1041,5 +1045,94 @@ gemmkernel_6x2_Avx2_fp16_fA0fB0fC0(GemmParamsFP16* gp) {
         "r15",
         "memory");
 }
+#else // _MSC_VER
+// Intrinsic kernel for MSVC
+void gemmkernel_Avx2_fp16_fA0fB0fC0(GemmParamsFP16* gp, const size_t kernel_nrows) {
+  // register buffer
+  __m256 ymmSum[12];
+  size_t idxA = 0, idxB = 0, idxC = 0;
+  // ldc in float size
+  size_t ldc_floatsize = gp->ldc / sizeof(float);
+  // load beta
+  __m256 ymmBeta;
+  if (gp->beta != 0)
+    ymmBeta = _mm256_broadcast_ss(&gp->beta);
+
+  // outer loop - block columns
+  for(uint64_t ii = 0; ii < gp->b_block_cols; ii++) {
+    // reset index
+    idxA = 0;
+    // inner loop - k
+    for(uint64_t kk = 0; kk < gp->k; kk++) {
+      // load B
+      __m256 ymmB0 = _mm256_cvtph_ps(_mm_load_si128((__m128i*)(gp->B + idxB)));
+      __m256 ymmB1 = _mm256_cvtph_ps(_mm_load_si128((__m128i*)(gp->B + idxB + 8)));
+      idxB += 16;
+
+      // first element
+      if (kk == 0) {
+        if(gp->beta != 0) {  // accumulate
+          for(size_t jj = 0; jj < kernel_nrows; jj++) {
+            // load A
+            __m256 ymmA = _mm256_broadcastss_ps(_mm_broadcast_ss((float const*)(gp->A + idxA + jj)));
+            // C = A * B + beta * C
+            ymmSum[2 * jj] = _mm256_fmadd_ps(ymmA, ymmB0, _mm256_mul_ps(ymmBeta, _mm256_loadu_ps(gp->C + idxC + jj * ldc_floatsize)));
```

Comment:
If it's not too much trouble, can we move this functions to a separate file because this file is generated by a Python script? Also doing clang formatter will be really appreciated.